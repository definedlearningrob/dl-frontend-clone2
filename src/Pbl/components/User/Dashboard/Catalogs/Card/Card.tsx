import CatalogCardContent from '@pbl/components/User/Dashboard/Catalogs/Card/CatalogCardContent';
import { TCatalog } from '@pbl/graphql/user/queries/dashboardCatalog';

type Props = {
  tasksCount?: number;
  tracksCount?: number;
  coursesCount?: number;
  projectsCount?: number;
  displayName?: string;
  id: string;
  selectedCatalogData: TCatalog;
  onClick: () => void;
  isSingleCard?: boolean;
};

const UserDashboardCatalogsCard = ({
  coursesCount,
  projectsCount,
  displayName,
  tasksCount,
  tracksCount,
  id,
  selectedCatalogData,
  onClick,
  isSingleCard,
}: Props) => {
  const primaryCount = isSingleCard ? tracksCount : coursesCount;
  const secondaryCount = isSingleCard ? tasksCount : projectsCount;

  return (
    <CatalogCardContent
      displayName={displayName}
      isActive={id === selectedCatalogData.id}
      primaryCount={primaryCount || 0}
      secondaryCount={secondaryCount || 0}
      onClick={onClick}
    />
  );
};

export default UserDashboardCatalogsCard;
