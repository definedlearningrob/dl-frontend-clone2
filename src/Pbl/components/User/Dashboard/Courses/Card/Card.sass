@import '@shared/stylesheets/config.sass'

$lg-height-content: 2 * 1.2 + rem
$card-image-height: 140px
$card-min-width: 200px
$card-max-width: 300px

.dashboard-course-card
  +flex-container(column)
  background-color: $color-white
  border-radius: $border-radius-xs
  border: $border-width-xs solid $color-border
  gap: $m-xs
  min-width: $card-min-width
  padding: $m-xs $m-xs $m-sm
  transition: box-shadow $transition-duration $transition-function

  &:hover
    box-shadow: $shadow-200

  &__image-wrapper
    +flex-container(row, center, center)
    border-radius: $border-radius-xs
    flex: 0 0 $card-image-height
    overflow: hidden

  &__image
    min-height: 100%
    min-width: 100%
    object-fit: cover

  &__name
    +ellipsis-multiline(2)
    font-size: $font-size-sm
    margin-bottom: $m-xs
    height: $lg-height-content
    line-height: 1.2

  &__description
    +flex-container(column)
    color: $color-font-secondary
    flex: auto
    height: $lg-height-content
    font-size: $font-size-xs
    margin-bottom: $m-xs

  &__content
    padding: 0 $m-xs

  &__short-description-inject
    +ellipsis-multiline(2)

    & *
      display: inline

  &__description-link
    font-size: $font-size-xs
    width: max-content
    margin: auto $m-xs 0

  &__description-link-icon
    margin-left: $m-xs

  &__grades-text
    white-space: nowrap
    font-size: $font-size-xxs
    margin-bottom: 0

  &__grade
    display: inline-block
    min-width: 24px
    text-align: center
    white-space: nowrap

  &__project-summary
    border-radius: $border-radius-lg
    border: $border-width-xs solid $color-background
    padding: $m-xxs $m-xs

.card-data
  +flex-container(row)
  padding: 0 $m-xs
