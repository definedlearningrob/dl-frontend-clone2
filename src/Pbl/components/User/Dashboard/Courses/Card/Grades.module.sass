@import '@shared/stylesheets/config.sass'

$grade-max-width: $m-base

.gradesWrapper
  border-radius: $border-radius-lg
  border: $border-width-xs solid $color-background
  margin-right: $m-xs
  padding: $m-xxs $m-xs
  min-width: 0
