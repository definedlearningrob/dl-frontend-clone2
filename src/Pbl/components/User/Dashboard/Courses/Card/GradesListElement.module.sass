@import '@shared/stylesheets/config.sass'

.grade
  text-align: center
  white-space: nowrap
  min-width: 0

.gradeText
  +ellipsis()
  font-size: $font-size-xxs
  margin-bottom: 0
  min-width: 0
