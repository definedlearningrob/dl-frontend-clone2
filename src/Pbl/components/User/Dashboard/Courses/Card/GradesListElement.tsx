import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './GradesListElement.module.sass';

type Props = {
  range: {
    start: string;
    end: string;
  };
};

export const GradesListElement = ({ range }: Props) => {
  const { t } = useTranslation();
  const getRangeString = (range: { start: string; end: string }) =>
    range.start === range.end ? range.start : `${range.start}-${range.end}`;
  const gradesRange = useMemo(() => getRangeString(range), []);

  return (
    <li className={styles.grade} data-testid='grade-list-element'>
      <p className={styles.gradeText} data-testid='grade-text'>
        {t('user.dashboard.gradesRange', { gradesRange })}
      </p>
    </li>
  );
};
