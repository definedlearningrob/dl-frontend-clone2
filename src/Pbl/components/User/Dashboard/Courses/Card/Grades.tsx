import { TGrade } from '@pbl/graphql/user/queries/dashboardCourses';
import { GradesListElement } from '@pbl/components/User/Dashboard/Courses/Card/GradesListElement';

import { groupGradesByRange } from '@shared/utils/groupGradesByRange';

import styles from './Grades.module.sass';

type Props = {
  grades: TGrade[];
};

export const Grades = ({ grades }: Props) => {
  const groupedGradesByRange = groupGradesByRange(grades);
  const firstGrade = groupedGradesByRange[0].start;
  const lastGrade = groupedGradesByRange[groupedGradesByRange.length - 1].end;
  const groupedGrades =
    groupedGradesByRange.length === 1
      ? groupedGradesByRange
      : [{ start: firstGrade, end: lastGrade }];

  const renderGrades = () =>
    groupedGrades.map((range) => <GradesListElement key={range.start} range={range} />);

  return (
    <ul className={styles.gradesWrapper} data-testid='grades-summary'>
      {renderGrades()}
    </ul>
  );
};
