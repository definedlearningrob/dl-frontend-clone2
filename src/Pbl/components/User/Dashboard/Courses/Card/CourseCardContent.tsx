/* eslint-disable react/no-danger */
import { isEmpty } from 'lodash-es';
import { useTranslation } from 'react-i18next';

import { TDashboardCoursesTrack, TGrade } from '@pbl/graphql/user/queries/dashboardCourses';

import SharedImage from '@shared/components/Image/Image';
import { cleanInjection } from '@shared/utils/cleanInjection';
import SharedButton from '@shared/components/Button/Button';
import SharedIcon from '@shared/components/Icon/Icon';
import { ReactComponent as ChevronRight } from '@shared/svg/chevron_right.svg';

import { Grades } from './Grades';
import '@pbl/components/User/Dashboard/Courses/Card/Card.sass';

type Props = {
  course: TDashboardCoursesTrack;
  grades: TGrade[];
  onClick: () => void;
};

const CourseCardContent = ({ course, onClick, grades }: Props) => {
  const { displayName, imageUrl, thumbnailUrl, tasksCount, shortDescription } = course;
  const { t } = useTranslation();

  return (
    <div className='dashboard-course-card'>
      <div className='dashboard-course-card__image-wrapper'>
        <SharedImage
          alt={displayName}
          className='dashboard-course-card__image'
          fallbackSrc={imageUrl}
          src={thumbnailUrl}
        />
      </div>
      <div className='card-data' data-testid='card-data'>
        {!isEmpty(grades) && <Grades grades={grades} />}
        <div className='dashboard-course-card__project-summary'>
          <p className='dashboard-course-card__grades-text'>
            {t('user.dashboard.projectsCount', { tasksCount })}
          </p>
        </div>
      </div>
      <div className='dashboard-course-card__content'>
        <h6 className='dashboard-course-card__name' data-testid='dashboard-course-item-name'>
          {displayName}
        </h6>
        <p className='dashboard-course-card__description'>
          <span
            className='dashboard-course-card__short-description-inject'
            dangerouslySetInnerHTML={cleanInjection(shortDescription)}
          />
        </p>
      </div>
      <SharedButton
        className='dashboard-course-card__description-link'
        data-testid='description-link'
        variant='link'
        onClick={onClick}>
        {t('user.dashboard.view')}
        <SharedIcon
          className='dashboard-course-card__description-link-icon'
          icon={<ChevronRight />}
          size='xs'
        />
      </SharedButton>
    </div>
  );
};

export default CourseCardContent;
