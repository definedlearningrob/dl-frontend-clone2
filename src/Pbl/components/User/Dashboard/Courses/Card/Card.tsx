import { useHistory } from 'react-router-dom';

import type { TDashboardCoursesTrack } from '@pbl/graphql/user/queries/dashboardCourses';
import '@pbl/components/User/Dashboard/Courses/Card/Card.sass';
import CourseCardContent from '@pbl/components/User/Dashboard/Courses/Card/CourseCardContent';

type Props = {
  course: TDashboardCoursesTrack;
};

export const Card = ({ course }: Props) => {
  const { id, grades } = course;
  const history = useHistory();

  const goToCourse = () => history.push(`/courses/${id}`);

  return <CourseCardContent course={course} grades={grades} onClick={goToCourse} />;
};
