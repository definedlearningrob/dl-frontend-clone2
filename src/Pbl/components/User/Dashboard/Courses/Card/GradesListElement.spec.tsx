import { renderWithRouter } from '@pbl/utils/test';
import { GradesListElement } from '@pbl/components/User/Dashboard/Courses/Card/GradesListElement';

describe('GradesListElement', () => {
  const mockedRange = {
    start: 'K',
    end: '5',
  };

  it('should display grades', () => {
    const { getByTestId, getByText } = renderWithRouter(<GradesListElement range={mockedRange} />);

    expect(getByTestId('grade-list-element')).toBeInTheDocument();
    expect(getByTestId('grade-text')).toBeInTheDocument();
    expect(getByText('Grades: K-5')).toBeInTheDocument();
  });

  it('should renders correctly', () => {
    // @ts-ignore
    const { container } = renderWithRouter(<GradesListElement range={mockedRange} />);

    expect(container).toMatchSnapshot();
  });
});
