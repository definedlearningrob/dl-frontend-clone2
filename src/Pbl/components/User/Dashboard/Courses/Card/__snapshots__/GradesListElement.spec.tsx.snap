// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradesListElement should renders correctly 1`] = `
<div>
  <li
    class="grade"
    data-testid="grade-list-element"
  >
    <p
      class="gradeText"
      data-testid="grade-text"
    >
      Grades: K-5
    </p>
  </li>
</div>
`;
